### DISCUSSION TITLE
Accessing list of configured categories

### DISCUSSION THREAD

**Discussion by Dekker500** [orestbida/cookieconsent#513] [Q&A]
Have everything configured and working great so far.

In one particular script, I am trying to walk through the list of configured categories. I can hard-code it, as I don't have that many, but I thought it would be much easier to access the "actual" configuration.

I thought something like this would work, but it is giving me "undefined".
```
// What is set in the user's current cookie
const cookieCategories = CC.getCookie('categories');
console.log(" CC.categories);
// what are the possible cookie categories that I have configured
const possibleCategories = Object.entries(CC.categories);
console.log(possibleCategories);
```

The CC.categores is returning "undefined" and is therefore failing. Is the CC.categories ONLY at configuration?

Is there an equivalent runtime API to access the POSSIBLE categories?

**Comment by orestbida** [comment]
You can retrieve categories via `.getConfig()`  or `getUserPreferences()`.

```javascript
const cc = initCookieConsent();

cc.run({...});

const categories = cc.getConfig('languages')['en'].settings_modal.blocks
    .filter(block => !!block.toggle)
    .map(block => block.toggle.value);

// or 
const preferences = cc.getUserPreferences();
const allCategories = [...preferences.accepted_categories, ...preferences.rejected_categories];
```
Replace `'en'` with a valid language code if necessary.
